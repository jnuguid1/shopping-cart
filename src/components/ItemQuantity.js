import React from 'react';

const ItemQuantity = (props) => {
  return (
    <div className='item-quantity'>
      <p>Quantity </p>
      <button 
        type='button'
        onClick={e => props.onQuantityDecrement(props.item, e)}
      >
        -
      </button>
      <input 
        value={props.item.input} 
        type='text'
        onChange={e => props.onQuantityChange(props.item, e)} 
      />
      <p data-testid='quantity'>{props.quantity}</p>
      <button
        onClick={e => props.onQuantityIncrement(props.item, e)}
        type='button'
      >
        +
      </button>
    </div>
  );
};

export default ItemQuantity;