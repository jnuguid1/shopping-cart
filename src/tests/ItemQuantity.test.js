import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ItemQuantity from '../components/ItemQuantity';

describe('Item quantity component', () => {
  it('renders 0 quantity', () => {
    const { container } = render(<ItemQuantity />);
    expect(container).toMatchSnapshot();
  })

  it('quantity increments on increment button click', () => {
    const onChangeMock = jest.fn();
    render(<ItemQuantity quantity={1} onQuantityIncrement={onChangeMock} />);
    const button = screen.getByRole('button', { name: '+' });
    userEvent.click(button);
    expect(screen.getByTestId('quantity').textContent).toMatch('2');
  })
})