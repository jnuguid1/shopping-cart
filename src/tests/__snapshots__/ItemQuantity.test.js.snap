// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item quantity component renders 0 quantity 1`] = `
<div>
  <div
    class="item-quantity"
  >
    <p>
      Quantity 
    </p>
    <button
      type="button"
    >
      -
    </button>
    <p
      data-testid="quantity"
    />
    <button
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
